SeaPublicWebsite.Data/Migrations/20220708114101_UpdateHeatingTypeColumnInEpc.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace SeaPublicWebsite.Data.Migrations
+{
+    public partial class UpdateHeatingTypeColumnInEpc : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
+                name: "EpcHeatingType",
+                table: "Epc",
+                type: "integer",
+                nullable: true);
+            
+            // Convert the other heating type into the new column
+            // Original column enum had 8 options, so we stack these new values
+            // starting from 8
+            // We also ignore the 'other' option since it already exists in the normal heating type
+            migrationBuilder.Sql(@"
+                UPDATE public.""Epc""
+                SET ""EpcHeatingType"" = ""Epc"".""OtherHeatingType"" + 8
+                WHERE ""Epc"".""OtherHeatingType"" IS NOT NULL AND ""Epc"".""OtherHeatingType"" != 2");
+            
+            // Then copy over the regular heating type values
+            migrationBuilder.Sql(@"
+                UPDATE public.""Epc""
+                SET ""EpcHeatingType"" = ""Epc"".""HeatingType""
+                WHERE ""Epc"".""HeatingType"" IS NOT NULL");
```

Comment:
Let me know if you want to have a chat about this one, it's not that straight forward (as is with any migration including merging two enum columns...)

Reply:
This makes sense thanks :)

We can get away with this while the site isn't live, but it's interesting to think about how we would do this seamlessly in the future.
To be explicit, the problem here is that we are running two servers in parallel. During deployment of the first server this migration is run. Until the second server is deployed it will not be able to understand the EPC table anymore and may throw errors.